PythonCodeConventions.md

```diff
@@ -49,6 +58,46 @@ y = fib(2.5)
 In this case, mypy will inform that the call using the floating point value is not valid:
 __mypytest.py:10: error: Argument 1 to "fib" has incompatible type "float"; expected "int"__
 
+The following construct is _only_ allowed in case of circular dependencies, otherwise use regular import statements.
+``` python
+MYPY = False
+if MYPY:
+    from [namespace] import [class]
+```
+Line breaking
+---
+Indentation of function definitions and function calls that exceed the maximum line length (PEP8: 80, Ultimaker: 120 characters) is described in PEP8 https://www.python.org/dev/peps/pep-0008/#indentation. From the suggested set, the following must be applied:
+
+When a function definition exceeds the maximum line length, wrap the lines as such:
+``` python
+def long_function_name(
+    var_one: str,
+    var_two: int,
+    var_three: float,
+    *,
+    var_four: bool=False
+) -> None:
```

Comment:
This is ambiguous. Should we always put *every* new parameter on a newline once it's longer? Because that isn't the case with the function call example below.

Reply:
This example is taken from PEP8, with some minor addittions (the boolean trap prevention and typing). The difference is in splitting up parameters for a function _definition_ and function _call_.  See: https://www.python.org/dev/peps/pep-0008/#indentation
I recall that we'd want to prevent spreading parameters for a long function call, because that would hamper reading the code of a single function. But with a function definition this would not be the case and putting each param on a new line, would contribute to readability, also because of the typing we prescribe.

Reply:
Well, then the official one for pep8 is ambiguous. 

As for adding bool traps, I'm all in favor of putting it in there, but make it explicit (eg; an actual rule and not only something to be gleaned from an example).

Reply:
The intention was not to prescribe boolean traps (or prevention thereof) but to provide an example with as much function definition elements as possible.

Reply:
And yes, PEP8 does not prescribe a single way of splitting up long lines.

Reply:
To be honest I really don't care that much about how people split lines as long as a little bit of common sense is applied. It is possible to define rules for this kind of thing, but most people can never remember what these "lesser used" rules are. Also just from my experience with auto-format functions in IDEs, rules for line splitting/wrapping usually produce awful and unreadable results compared to someone just using their brain for each situation.

Reply:
Problem is that what is considered 'common sense' differs per reviewer. The specific newly added point where discussed because we had trouble agreeing and think we would benefit from a/any specific choice. This will help/support uniform reviews opposed to different comments from different reviewers.

Reply:
Also; not suggesting auto formatting either.

Reply:
To put it bluntly, I don't believe we can a) make a rule that people can remember, and b) make a rule which gives good readable results most of the time.

Reply:
I agree with @sedwards2009. The biggest problem is making a rule that is both sensible and will be actually remembered (and subsequently; used).

Reply:
Well, this _is_ taken from a bunch of PEP8 proposed solutions, so whether they are taken out of this document, they are still valid and should/could be used/applied. So that would make them sensible, because we adhere to PEP8. The problem is that PEP8 offers more than one solution, it is suggested to make a specific choice in the matter.

Reply:
Applied @Ghostkeeper 's suggestion